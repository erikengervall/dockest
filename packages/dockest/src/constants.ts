export const LOG_LEVEL = {
  NOTHING: 0,
  ERROR: 1,
  WARN: 2,
  INFO: 3,
  DEBUG: 4,
}

export const GENERATED_COMPOSE_FILE_PATH = `${process.cwd()}/docker-compose.dockest-generated.yml`

export const DOCKEST_ATTACH_TO_PROCESS = 'DOCKEST_ATTACH_TO_PROCESS'

export const BRIDGE_NETWORK_NAME = `dockest_bridge_network`

export const DOCKEST_HOST_ADDRESS = 'host.dockest-runner.internal'
export const DEFAULT_HOST_NAME = 'host.docker.internal'

export const DEFAULT_OPTS = {
  composeFile: 'docker-compose.yml',
  jestOpts: {
    projects: ['.'],
    runInBand: true,
  },
  logLevel: LOG_LEVEL.INFO,
}

export const DEFAULT_$ = {
  dockestServices: [],
  hostname: process.env.HOSTNAME || DEFAULT_HOST_NAME,
  isInsideDockerContainer: false,
  jestRanWithResult: false,
  perfStart: 0,
  runners: [],
}

/**
 * Released 2017-05-06: https://github.com/facebook/jest/releases/tag/v20.0.0
 */
export const MINIMUM_JEST_VERSION = '20.0.0'
