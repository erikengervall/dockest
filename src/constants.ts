export const COLORS = {
  MISC: {
    BLINK: '\x1b[5m',
    BRIGHT: '\x1b[1m',
    DIM: '\x1b[2m',
    HIDDEN: '\x1b[8m',
    RESET: '\x1b[0m',
    REVERSE: '\x1b[7m',
    UNDERSCORE: '\x1b[4m',
  },
  FG: {
    BLACK: '\x1b[30m',
    BLUE: '\x1b[34m',
    CYAN: '\x1b[36m',
    GREEN: '\x1b[32m',
    MAGENTA: '\x1b[35m',
    RED: '\x1b[31m',
    WHITE: '\x1b[37m',
    YELLOW: '\x1b[33m',
  },
  BG: {
    BLACK: '\x1b[40m',
    BLUE: '\x1b[44m',
    CYAN: '\x1b[46m',
    GREEN: '\x1b[42m',
    MAGENTA: '\x1b[45m',
    RED: '\x1b[41m',
    WHITE: '\x1b[47m',
    YELLOW: '\x1b[43m',
  },
}

export const LOG_SYMBOLS: readonly string[] = [
  'üêâ ',
  'üêí ',
  'üêô ',
  'üêû ',
  'üê• ',
  'üêº ',
  'üêø ',
  'ü¶Ç ',
  'ü¶É ',
  'ü¶Ñ ',
  'ü¶ä ',
  'ü¶ã ',
  'ü¶ç ',
  'ü¶ñ ',
  'ü¶ö ',
]

export const LOG_LEVEL = {
  NOTHING: 0,
  ERROR: 1,
  WARN: 2,
  INFO: 3,
  DEBUG: 4,
}

export const DEFAULT_USER_CONFIG = {
  afterSetupSleep: 0,
  dev: { debug: false },
  dumpErrors: false,
  exitHandler: null,
  logLevel: LOG_LEVEL.INFO,
  runInBand: true,
  composeFile: [],
  guessRunnerType: false,
}

export const INTERNAL_CONFIG = {
  failedTeardowns: [],
  jestRanWithResult: false,
  perfStart: 0,
  isInsideDockerContainer: false,
}

export const PROCESS_TEST_ENV = 'dockest-test'

export const GENERATED_COMPOSE_FILE_NAME = 'docker-compose.dockest-generated.yml'
export const GENERATED_RUNNER_COMPOSE_FILE_NAME = 'docker-compose.dockest-generated-runner.yml'
export const GENERATED_COMPOSE_FILE_PATH = `${process.cwd()}/${GENERATED_COMPOSE_FILE_NAME}`
export const GENERATED_RUNNER_COMPOSE_FILE_PATH = `${process.cwd()}/${GENERATED_RUNNER_COMPOSE_FILE_NAME}`

export const BRIDGE_NETWORK_NAME = `dockest_bridge_network`
