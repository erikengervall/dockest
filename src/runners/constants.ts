import { SharedDefaultableConfigProps } from './@types'

export const SHARED_DEFAULT_CONFIG_PROPS: SharedDefaultableConfigProps = {
  build: '',
  commands: [],
  connectionTimeout: 5,
  dependsOn: [],
  host: 'localhost',
  image: undefined,
  networks: undefined,
  ports: [],
  props: {},
  responsivenessTimeout: 30,
}
