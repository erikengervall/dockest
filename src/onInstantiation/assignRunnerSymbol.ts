import { DockestConfig } from '../index'
import { LOG_SYMBOLS } from '../constants'
import hashCode from '../utils/hashCode'

export default (config: DockestConfig) => {
  let LOG_SYMBOLS_CLONE = LOG_SYMBOLS.slice(0)

  config.runners.forEach(({ logger, runnerConfig: { service } }) => {
    const serviceHash = Math.abs(hashCode(service))

    if (LOG_SYMBOLS_CLONE.length === 0) {
      LOG_SYMBOLS_CLONE = LOG_SYMBOLS.slice(0) // If there's not enough symbols, reinitialize the array
    }

    const index = serviceHash % LOG_SYMBOLS_CLONE.length
    const LOG_SYMBOL = LOG_SYMBOLS_CLONE.splice(index, 1)[0]

    logger.setRunnerSymbol(LOG_SYMBOL)
  })
}
