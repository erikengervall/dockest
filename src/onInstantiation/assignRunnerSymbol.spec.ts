import { createMockProxy } from 'jest-mock-proxy'
import { DEFAULT_USER_CONFIG, INTERNAL_CONFIG, LOG_SYMBOLS } from '../constants'
import { runners } from '../index'
import assignRunnerSymbol from './assignRunnerSymbol'

jest.mock('../Logger')

const { RedisRunner } = runners
const createDockestConfig = (numberOfRunners: number = 1) => {
  const manyRedisRunners = Array.from(Array(numberOfRunners), (_, index) => {
    const redisRunner = new RedisRunner({ service: `${index}`, image: '_' })
    redisRunner.logger = createMockProxy()
    return redisRunner
  })

  return {
    runners: manyRedisRunners,
    opts: DEFAULT_USER_CONFIG,
    jest: {},
    $: INTERNAL_CONFIG,
  }
}

describe('assignRunnerSymbol', () => {
  it('should handle the base case and reset the symbols', () => {
    const numberOfRunner = LOG_SYMBOLS.length + 1
    const dockestConfig = createDockestConfig(numberOfRunner)

    assignRunnerSymbol(dockestConfig)

    for (const runner of dockestConfig.runners) {
      expect(runner.logger.setRunnerSymbol).toHaveBeenCalledWith(
        expect.stringMatching(new RegExp(LOG_SYMBOLS.join('|'), 'gi')),
      )
    }
  })
})
