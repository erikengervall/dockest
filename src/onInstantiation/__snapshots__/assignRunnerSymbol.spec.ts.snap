// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignRunnerSymbol should handle the base case and reset the symbols 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐞 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦃 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦖 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐥 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 5`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦋 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 6`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐼 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 7`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐉 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 8`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦚 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 9`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐒 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 10`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦄 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 11`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦂 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 12`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐙 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 13`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🐿 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 14`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦊 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`assignRunnerSymbol should handle the base case and reset the symbols 15`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "🦍 ",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
